modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableViewSkinTest.java

```diff
@@ -188,6 +189,43 @@ void testColumnHeaderReorderCorrectTranslateX() {
         assertEquals(20, columnDragHeader.getTranslateX());
     }
 
+    @Test
+    void testHeaderReorderWithinNestedColumns() {
+        TableView<String> tableView = new TableView<>();
+        for (int i = 0; i < 2; i++) {
+            TableColumn<String, String> column = new TableColumn<>("Col " + i);
+            column.setMinWidth(100);
+            column.setMaxWidth(100);
+            tableView.getColumns().add(column);
+        }
+
+        TableColumn<String, String> column = new TableColumn<>("Column with nested");
+        for (int i = 0; i < 2; i++) {
+            TableColumn<String, String> nestedCol = new TableColumn<>("NestedCol " + i);
+            nestedCol.setMinWidth(100);
+            nestedCol.setMaxWidth(100);
+            column.getColumns().add(nestedCol);
+        }
+        tableView.getColumns().add(column);
+
+        stageLoader = new StageLoader(tableView);
+
+        TableHeaderRow header = (TableHeaderRow) tableView.lookup("TableHeaderRow");
+        Node columnDragHeader = header.lookup(".column-drag-header");
+
+        assertEquals(0, columnDragHeader.getTranslateX());
+
+        NestedTableColumnHeader nestedTableColumnHeader =
+                (NestedTableColumnHeader) header.getRootHeader().getColumnHeaders().get(2);
+        TableColumnHeader tableColumnHeader = nestedTableColumnHeader.getColumnHeaders().get(0);
+
+        Bounds bounds = tableColumnHeader.localToScene(tableColumnHeader.getLayoutBounds());
+        TableColumnHeaderShim.columnReordering(tableColumnHeader, bounds.getMinX() + 20, bounds.getMinY());
+
+        // 200, since we have 2 columns to the left with a size of 100.
```

Comment:
Can you also add why assertion is done for 220 in this comment?

Reply:
good point.

May I suggest to declare constants for column widths and column reordering coordinates so the origin of these numbers becomes clear?
Also, we are comparing floating point numbers, so perhaps it's better specify a small delta:
```
double CWIDTH = 100;
double EPSILON = 0.0001;
assertEquals(CWIDTH + CWIDTH + OFFSET, columnDragHeader.getTranslateX(), EPSILON);
```

Reply:
Pushed. 
Since we compare very simple numbers (we even set them before as int), I personally would not expect a delta but an accurate number, therefore no delta added in this test.

Reply:
Will it be a round integer in case of fractional scale (it might be in this case, since the test does not set the scale explicitly)?

Should we extends this test with fractional scale, let's say 1.25?

Have you tested this fix with fractional scale?